apps/web/src/blocks/TextImageBlock/index.tsx

```diff
@@ -1,101 +1,243 @@
+      max-width: 100%;
+      height: auto;
     }
-  `};
-`;
 
-const ImageWrapper = s(ResponsivePayloadImage)`
-  aspect-ratio: 500 / 402;
-  position: relative;
+    ${($layout === 'imgLeftCenter' || $layout === 'imgLeft') &&
+    css`
+      margin-left: auto;
+    `}
 
-  img {
-    position: absolute;
-    width: 100%;
-    height: 100%;
-    object-fit: cover;
-  }
+    ${$layout === 'imgRight' &&
+    css`
+      margin-right: auto;
+    `}
+  `}
+`;
+
+const VideoWrapper = s.div<{ $layout: string }>`
+  ${({ $layout }) => css`
+    display: flex;
+    grid-area: image;
+    overflow: hidden;
+    align-self: center;
+    ${($layout === 'imgLeftCenter' || $layout === 'imgLeft') &&
+    css`
+      margin-left: auto;
+    `}
+
+    ${$layout === 'imgRight' &&
+    css`
+      margin-right: auto;
+    `}
+  `}
 `;
 
-const ContentWrapper = styled.div`
+const ButtonWrapper = s.div`
   display: flex;
-  flex-direction: column;
-  justify-content: center;
-  gap: 1.25rem;
-  width: min(100%, 476px);
+  gap: 1rem;
+`;
+
+const InputWrapper = s(FormWrapper)`
+  form {
+    display: flex;
+    gap: 1rem;
+    button { 
+      height: fit-content;
+      align-self: flex-end;
+    }
+  }
 `;
 
-const Title = styled.h3({ m: 0, p: 0 })``;
+const defaultImageProps = (image: Image | number) => {
+  if (typeof image === 'number') {
+    return image;
+  }
+
+  return {
+    ...image,
+    image,
+    imageProps: {
+      fill: false
+    }
+  };
+};
 
 function TextImageBlock({
-  title,
   content,
+  layout,
   image,
+  items,
+  blockConfig,
+  textinput,
   cta,
-  blockConfig
+  video
 }: TextImageBlockType) {
-  const layout = blockConfig?.layout || 'imgRight';
-  const className = genClassName([layout]);
+  const buttonLayout = !!items?.length || !!cta || false;
+  const imgLayout = layout || 'imgRight';
+
+  const Items = useMemo(() => {
+    if (!items || !items.length) {
+      return null;
+    }
+
+    return items.map(({ cta: t, id }) => {
+      if (!t) {
+        return null;
+      }
+      return (
+        <span key={id}>
+          {t?.link?.label && <CtaButton cta={t} color="primary" />}
+        </span>
+      );
+    });
+  }, [items]);
   return (
-    <Wrapper>
-      <InnerWrapper className={className}>
-        <ImageWrapper image={image} />
-        <ContentWrapper>
-          {title && <Title>{title}</Title>}
-          {content && <RichText {...content} />}
-          {cta && <CtaButton cta={cta} />}
+    <Wrapper {...blockConfig} hidden={blockConfig?.hidden ?? false}>
+      <BlockWrapper $layout={imgLayout}>
+        <ContentWrapper $hasButton={buttonLayout || false} $layout={imgLayout}>
+          {content && <Content $layout={imgLayout} {...content} />}
+          {items && <ButtonWrapper>{Items}</ButtonWrapper>}
+          {textinput && cta && (
+            <InputWrapper onSubmit={(data) => console.log(data)} cta={cta}>
+              <TextInput
+                {...textinput}
+                name="TextInput"
```

Comment:
Hardcoding for these values because the typing for these fields needs to be updated in refract. The label and placeholder are incompatible with the typing for payload text strings